PSAP-1003: NTO: reapply_sysctl and verify Profiles improvements (#56773)
openshift/openshift-docs: modules/custom-tuning-example.adoc

Tuned CR to use OpenShift-specific tuning for compute nodes.

// Note the issues with including profiles sharing the same ancestor: see link:https://bugzilla.redhat.com/show_bug.cgi?id=1825882[BZ#1825882]

*Overriding host-level sysctls*

> Various kernel parameters can be changed at runtime via `/run/sysctl.d/`, `/etc/sysctl.d/` and `/etc/sysctl.conf` host configuration files.  The {product-title} itself adds several host configuration files which set kernel parameters at runtime. Most notably `net.ipv[4-6].`, `fs.inotify.` and `vm.max_map_count`. These runtime parameters provide very basic functional tuning for the system prior to the kubelet and the Operator start.

Reviewer: ```suggestion
Various kernel parameters can be changed at runtime by using `/run/sysctl.d/`, `/etc/sysctl.d/`, and `/etc/sysctl.conf` host configuration files. {product-title} adds several host configuration files which set kernel parameters at runtime; for example, `net.ipv[4-6].`, `fs.inotify.`, and `vm.max_map_count`. These runtime parameters provide basic functional tuning for the system prior to the kubelet and the Operator start.
```

> The Operator respects these settings and will not override them, unless the `reapply_sysctl` option is set to `false`. This will result in `TuneD` not applying the settings from the host configuration files after it applies its custom profile.

Reviewer: ```suggestion
The Operator does not override these settings unless the `reapply_sysctl` option is set to `false`. Setting this option to `false` results in `TuneD` not applying the settings from the host configuration files after it applies its custom profile.
```

Reviewer: I think my suggestion is accurate but I wasn't 100% sure what "this" referred to - setting the option to false, or overriding the settings, or something else.

Author: Your suggestion is accurate.  Thank you.  Accepted.

Tuned CR to use OpenShift-specific tuning for compute nodes.

// Note the issues with including profiles sharing the same ancestor: see link:https://bugzilla.redhat.com/show_bug.cgi?id=1825882[BZ#1825882]

*Overriding host-level sysctls*

> Various kernel parameters can be changed at runtime via `/run/sysctl.d/`,

Reviewer 2: Remove hard wrapping for this text as well please.

`/etc/sysctl.d/` and `/etc/sysctl.conf` host configuration files.
The {product-title} itself adds several host configuration files
which set kernel parameters at runtime. Most notably `net.ipv[4-6].`,

> `fs.inotify.` and `vm.max_map_count`. These runtime parameters provide a
> very basic functionality tuning for the system prior to the kubelet and the
> Operator start.

Reviewer 3: ```suggestion
`fs.inotify.` and `vm.max_map_count`. These runtime parameters provide
very basic functional tuning for the system prior to the kubelet and the
Operator start.
```

The Operator respects these settings and will not override them, unless the

> `reapply_sysctl` option is set to `false`.  This will result in `TuneD` not

Reviewer 2: There's two spaces after the period - delete one space.

Author: Cannot teach an old dog new tricks.  Always do two spaces after an end of a sentence.  Removed here as requested then.
